`npm install` fails with NPM 7

# 🐞 Bug report

### Command (mark with an `x`)
<!-- Can you pin-point the command or commands that are effected by this bug? -->
<!-- ✍️edit: -->

- [x] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc


### Is this a regression?

<!-- Did this behavior use to work in the previous version? -->

This bug did not occur a couple of days ago. It doesn't seem like it's caused by a specific CLI version. It might be caused by some (peer) dependency of a brand-new Angular project which changed its dependencies. 


### Description

Creating a new Angular project with the Angular CLI (`ng new`) fails when using the latest version of npm (7.5.x).

Error message see below.

## 🔬 Minimal Reproduction

1) Install latest version of Angular CLI: `npm install -g @angular/cli@latest`
2) Install latest version of npm: `npm install -g npm@latest`
3) Create a new Angular project: `ng new some-app`

Step 3 crashes with the below error message.

## 🔥 Exception or Error
<pre><code>
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: ng-npm@0.0.0
npm ERR! Found: @angular/compiler-cli@11.2.0-next.0
npm ERR! node_modules/@angular/compiler-cli
npm ERR!   dev @angular/compiler-cli@"~11.2.0-next.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/compiler-cli@"^11.0.0 || ^11.1.0-next" from @angular-devkit/build-angular@0.1102.0-next.0
npm ERR! node_modules/@angular-devkit/build-angular
npm ERR!   dev @angular-devkit/build-angular@"~0.1102.0-next.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/maximilianschwarzmuller/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/maximilianschwarzmuller/.npm/_logs/2021-02-03T21_29_25_647Z-debug.log
✖ Package install failed, see above.
The Schematic workflow failed. See above.
</code></pre>


## 🌍 Your Environment
<pre><code>
Angular CLI: 11.1.2
Node: 15.4.0
OS: darwin x64

Angular: undefined
... 
Ivy Workspace: <error>

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1101.2
@angular-devkit/core         11.1.2
@angular-devkit/schematics   11.1.2
@angular/cli                 11.1.2
@schematics/angular          11.1.2
@schematics/update           0.1101.2

</code></pre>

I've encountered similar issues (NPM 7.5.2), but when creating new project (`ng new ng-app`), I get:

```
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: ng-app@0.0.0
npm ERR! Found: typescript@4.1.3
npm ERR! node_modules/typescript
npm ERR!   dev typescript@"~4.1.2" from the root project
npm ERR!   peer typescript@"~4.0.0 || ~4.1.0" from @angular-devkit/build-angular@0.1101.2
npm ERR!   node_modules/@angular-devkit/build-angular
npm ERR!     dev @angular-devkit/build-angular@"~0.1101.2" from the root project
npm ERR!   2 more (@angular/compiler-cli, ng-packagr)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer typescript@"~3.9.5" from tsickle@0.39.1
npm ERR! node_modules/tsickle
npm ERR!   peerOptional tsickle@"~0.39.0" from ng-packagr@11.1.3
npm ERR!   node_modules/ng-packagr
npm ERR!     peerOptional ng-packagr@"^11.0.0 || ^11.1.0-next" from @angular-devkit/build-angular@0.1101.2
npm ERR!     node_modules/@angular-devkit/build-angular
npm ERR!       dev @angular-devkit/build-angular@"~0.1101.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

And when trying to `npm install` in my current project
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: project@1.6.4
npm ERR! Found: @angular/compiler@11.0.9
npm ERR! node_modules/@angular/compiler
npm ERR!   @angular/compiler@"~11.0.9" from the root project
npm ERR!   peer @angular/compiler@"11.0.9" from @angular/compiler-cli@11.0.9
npm ERR!   node_modules/@angular/compiler-cli
npm ERR!     dev @angular/compiler-cli@"~11.0.9" from the root project
npm ERR!     peer @angular/compiler-cli@"^11.0.0" from @angular-devkit/build-angular@0.1100.7
npm ERR!     node_modules/@angular-devkit/build-angular
npm ERR!       dev @angular-devkit/build-angular@"0.1100.7" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/compiler@"11.1.2" from @angular/localize@11.1.2
npm ERR! node_modules/@angular/localize
npm ERR!   peerOptional @angular/localize@"^11.0.0" from @angular-devkit/build-angular@0.1100.7
npm ERR!   node_modules/@angular-devkit/build-angular
npm ERR!     dev @angular-devkit/build-angular@"0.1100.7" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Both these were working just fine with NPM 6.x. But just looking at these errors, peer dependencies seem to mismanaged.

Environment:
```
Angular CLI: 11.1.2
Node: 14.15.1
OS: darwin x64

Angular: <error>
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1101.2 (cli-only)
@angular-devkit/build-angular   <error>
@angular-devkit/core            11.1.2 (cli-only)
@angular-devkit/schematics      11.1.2 (cli-only)
@angular/cli                    11.1.2 (cli-only)
@schematics/angular             11.1.2 (cli-only)
@schematics/update              0.1101.2 (cli-only)
rxjs                            6.6.3 (cli-only)
typescript                      <error>
```

I've encountered the same error
```
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: pwa@0.0.0
npm ERR! Found: typescript@4.1.3
npm ERR! node_modules/typescript
npm ERR!   dev typescript@"~4.1.2" from the root project
npm ERR!   peer typescript@"~4.0.0 || ~4.1.0" from @angular-devkit/build-angular@0.1101.2
npm ERR!   node_modules/@angular-devkit/build-angular
npm ERR!     dev @angular-devkit/build-angular@"~0.1101.2" from the root project
npm ERR!   2 more (@angular/compiler-cli, ng-packagr)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer typescript@"~3.9.5" from tsickle@0.39.1
npm ERR! node_modules/tsickle
npm ERR!   peerOptional tsickle@"~0.39.0" from ng-packagr@11.1.3
npm ERR!   node_modules/ng-packagr
npm ERR!     peerOptional ng-packagr@"^11.0.0 || ^11.1.0-next" from @angular-devkit/build-angular@0.1101.2
npm ERR!     node_modules/@angular-devkit/build-angular
npm ERR!       dev @angular-devkit/build-angular@"~0.1101.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

For more information, the json:
```json
{
  "code": "ERESOLVE",
  "current": {
    "name": "typescript",
    "version": "4.1.3",
    "whileInstalling": {
      "name": "pwa",
      "version": "0.0.0",
      "path": "/home/saadi/Coding/pwa"
    },
    "location": "node_modules/typescript",
    "dependents": [
      {
        "type": "dev",
        "name": "typescript",
        "spec": "~4.1.2",
        "from": {
          "location": "/home/saadi/Coding/pwa"
        }
      },
      {
        "type": "peer",
        "name": "typescript",
        "spec": "~4.0.0 || ~4.1.0",
        "from": {
          "name": "@angular-devkit/build-angular",
          "version": "0.1101.2",
          "whileInstalling": {
            "name": "pwa",
            "version": "0.0.0",
            "path": "/home/saadi/Coding/pwa"
          },
          "location": "node_modules/@angular-devkit/build-angular",
          "dependents": [
            {
              "type": "dev",
              "name": "@angular-devkit/build-angular",
              "spec": "~0.1101.2",
              "from": {
                "location": "/home/saadi/Coding/pwa"
              }
            }
          ]
        }
      },
      {
        "type": "peer",
        "name": "typescript",
        "spec": ">=4.0 <4.2",
        "from": {
          "name": "@angular/compiler-cli",
          "version": "11.1.2",
          "whileInstalling": {
            "name": "pwa",
            "version": "0.0.0",
            "path": "/home/saadi/Coding/pwa"
          },
          "location": "node_modules/@angular/compiler-cli",
          "dependents": [
            {
              "type": "dev",
              "name": "@angular/compiler-cli",
              "spec": "~11.1.1",
              "from": {
                "location": "/home/saadi/Coding/pwa"
              }
            },
            {
              "type": "peer",
              "name": "@angular/compiler-cli",
              "spec": "^11.0.0 || ^11.1.0-next",
              "from": {
                "name": "@angular-devkit/build-angular",
                "version": "0.1101.2",
                "whileInstalling": {
                  "name": "pwa",
                  "version": "0.0.0",
                  "path": "/home/saadi/Coding/pwa"
                },
                "location": "node_modules/@angular-devkit/build-angular",
                "dependents": [
                  {
                    "type": "dev",
                    "name": "@angular-devkit/build-angular",
                    "spec": "~0.1101.2",
                    "from": {
                      "location": "/home/saadi/Coding/pwa"
                    }
                  }
                ]
              }
            },
            {
              "type": "peer",
              "name": "@angular/compiler-cli",
              "spec": "11.1.2",
              "from": {
                "name": "@angular/localize",
                "version": "11.1.2",
                "whileInstalling": {
                  "name": "pwa",
                  "version": "0.0.0",
                  "path": "/home/saadi/Coding/pwa"
                },
                "location": "node_modules/@angular/localize",
                "dependents": [
                  {
                    "type": "peerOptional",
                    "name": "@angular/localize",
                    "spec": "^11.0.0 || ^11.1.0-next",
                    "from": {
                      "name": "@angular-devkit/build-angular",
                      "version": "0.1101.2",
                      "whileInstalling": {
                        "name": "pwa",
                        "version": "0.0.0",
                        "path": "/home/saadi/Coding/pwa"
                      },
                      "location": "node_modules/@angular-devkit/build-angular",
                      "dependents": [
                        {
                          "type": "dev",
                          "name": "@angular-devkit/build-angular",
                          "spec": "~0.1101.2",
                          "from": {
                            "location": "/home/saadi/Coding/pwa"
                          }
                        }
                      ]
                    }
                  }
                ]
              }
            },
            {
              "type": "peer",
              "name": "@angular/compiler-cli",
              "spec": "^11.0.0",
              "from": {
                "name": "ng-packagr",
                "version": "11.1.3",
                "whileInstalling": {
                  "name": "pwa",
                  "version": "0.0.0",
                  "path": "/home/saadi/Coding/pwa"
                },
                "location": "node_modules/ng-packagr",
                "dependents": [
                  {
                    "type": "peerOptional",
                    "name": "ng-packagr",
                    "spec": "^11.0.0 || ^11.1.0-next",
                    "from": {
                      "name": "@angular-devkit/build-angular",
                      "version": "0.1101.2",
                      "whileInstalling": {
                        "name": "pwa",
                        "version": "0.0.0",
                        "path": "/home/saadi/Coding/pwa"
                      },
                      "location": "node_modules/@angular-devkit/build-angular",
                      "dependents": [
                        {
                          "type": "dev",
                          "name": "@angular-devkit/build-angular",
                          "spec": "~0.1101.2",
                          "from": {
                            "location": "/home/saadi/Coding/pwa"
                          }
                        }
                      ]
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      {
        "type": "peer",
        "name": "typescript",
        "spec": ">=4.0 <4.2",
        "from": {
          "name": "ng-packagr",
          "version": "11.1.3",
          "whileInstalling": {
            "name": "pwa",
            "version": "0.0.0",
            "path": "/home/saadi/Coding/pwa"
          },
          "location": "node_modules/ng-packagr",
          "dependents": [
            {
              "type": "peerOptional",
              "name": "ng-packagr",
              "spec": "^11.0.0 || ^11.1.0-next",
              "from": {
                "name": "@angular-devkit/build-angular",
                "version": "0.1101.2",
                "whileInstalling": {
                  "name": "pwa",
                  "version": "0.0.0",
                  "path": "/home/saadi/Coding/pwa"
                },
                "location": "node_modules/@angular-devkit/build-angular",
                "dependents": [
                  {
                    "type": "dev",
                    "name": "@angular-devkit/build-angular",
                    "spec": "~0.1101.2",
                    "from": {
                      "location": "/home/saadi/Coding/pwa"
                    }
                  }
                ]
              }
            }
          ]
        }
      }
    ]
  },
  "edge": {
    "type": "peer",
    "name": "typescript",
    "spec": "~3.9.5",
    "error": "INVALID",
    "from": {
      "name": "tsickle",
      "version": "0.39.1",
      "whileInstalling": {
        "name": "pwa",
        "version": "0.0.0",
        "path": "/home/saadi/Coding/pwa"
      },
      "location": "node_modules/tsickle",
      "dependents": [
        {
          "type": "peerOptional",
          "name": "tsickle",
          "spec": "~0.39.0",
          "from": {
            "name": "ng-packagr",
            "version": "11.1.3",
            "whileInstalling": {
              "name": "pwa",
              "version": "0.0.0",
              "path": "/home/saadi/Coding/pwa"
            },
            "location": "node_modules/ng-packagr",
            "dependents": [
              {
                "type": "peerOptional",
                "name": "ng-packagr",
                "spec": "^11.0.0 || ^11.1.0-next",
                "from": {
                  "name": "@angular-devkit/build-angular",
                  "version": "0.1101.2",
                  "whileInstalling": {
                    "name": "pwa",
                    "version": "0.0.0",
                    "path": "/home/saadi/Coding/pwa"
                  },
                  "location": "node_modules/@angular-devkit/build-angular",
                  "dependents": [
                    {
                      "type": "dev",
                      "name": "@angular-devkit/build-angular",
                      "spec": "~0.1101.2",
                      "from": {
                        "location": "/home/saadi/Coding/pwa"
                      }
                    }
                  ]
                }
              }
            ]
          }
        }
      ]
    }
  },
  "peerConflict": null,
  "strictPeerDeps": false,
  "force": false
}
```

Hi all, 

**The first error:**
```
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/compiler-cli@"^11.0.0 || ^11.1.0-next" from @angular-devkit/build-angular@0.1102.0-next.0
npm ERR! node_modules/@angular-devkit/build-angular
npm ERR!   dev @angular-devkit/build-angular@"~0.1102.0-next.0" from the root project
```

Should be solved via https://github.com/angular/angular-cli/pull/19955

**The second error:**
```
npm ERR! Could not resolve dependency:
npm ERR! peer typescript@"~3.9.5" from tsickle@0.39.1
npm ERR! node_modules/tsickle
npm ERR!   peerOptional tsickle@"~0.39.0" from ng-packagr@11.1.3
npm ERR!   node_modules/ng-packagr
npm ERR!     peerOptional ng-packagr@"^11.0.0 || ^11.1.0-next" from @angular-devkit/build-angular@0.1101.2
npm ERR!     node_modules/@angular-devkit/build-angular
npm ERR!       dev @angular-devkit/build-angular@"~0.1101.2" from the root project
```

Is an NPM bug, kindly see https://github.com/npm/cli/issues/2615, as a workaround we can try to use the `--force` option.

I guess below is yet another NPM bug, but just FYI once a new Angular release is published, it is impossible to `ng new`/install  (e.g. after [adding a dependency](https://github.com/ng-bootstrap/ng-bootstrap/issues/3970)) a project with any of the older releases. I.e.

```
$ npm i -g @angular/cli@11.0
$ ng new hello
...
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: hello@0.0.0
npm ERR! Found: @angular/compiler@11.0.9
npm ERR! node_modules/@angular/compiler
npm ERR!   @angular/compiler@"~11.0.9" from the root project
npm ERR!   peer @angular/compiler@"11.0.9" from @angular/compiler-cli@11.0.9
npm ERR!   node_modules/@angular/compiler-cli
npm ERR!     dev @angular/compiler-cli@"~11.0.9" from the root project
npm ERR!     peer @angular/compiler-cli@"^11.0.0" from @angular-devkit/build-angular@0.1100.7
npm ERR!     node_modules/@angular-devkit/build-angular
npm ERR!       dev @angular-devkit/build-angular@"~0.1100.7" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/compiler@"11.1.2" from @angular/localize@11.1.2
npm ERR! node_modules/@angular/localize
npm ERR!   peerOptional @angular/localize@"^11.0.0" from @angular-devkit/build-angular@0.1100.7
npm ERR!   node_modules/@angular-devkit/build-angular
npm ERR!     dev @angular-devkit/build-angular@"~0.1100.7" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/lib/packages.ts
+++ b/lib/packages.ts
@@ -88,7 +88,7 @@ function loadPackageJson(p: string) {
       case 'engines':
         pkg['engines'] = {
           'node': '>= 12.13.0',
-          'npm': '^6.11.0',
+          'npm': '^6.11.0 || ^7.5.6',
           'yarn': '>= 1.13.0',
         };
         break;

--- a/packages/angular/cli/utilities/package-manager.ts
+++ b/packages/angular/cli/utilities/package-manager.ts
@@ -8,6 +8,7 @@
 import { execSync } from 'child_process';
 import { existsSync } from 'fs';
 import { join } from 'path';
+import { satisfies, valid } from 'semver';
 import { PackageManager } from '../lib/config/schema';
 import { getConfiguredPackageManager } from './config';
 
@@ -56,7 +57,7 @@ export async function getPackageManager(root: string): Promise<PackageManager> {
 }
 
 /**
- * Checks if the npm version is version 6.x.  If not, display a message and exit.
+ * Checks if the npm version is a supported 7.x version.  If not, display a warning.
  */
 export async function ensureCompatibleNpm(root: string): Promise<void> {
   if ((await getPackageManager(root)) !== PackageManager.Npm) {
@@ -64,21 +65,19 @@ export async function ensureCompatibleNpm(root: string): Promise<void> {
   }
 
   try {
-    const version = execSync('npm --version', {encoding: 'utf8', stdio: 'pipe'}).trim();
-    const major = Number(version.match(/^(\d+)\./)?.[1]);
-    if (major <= 6) {
+    const versionText = execSync('npm --version', {encoding: 'utf8', stdio: 'pipe'}).trim();
+    const version = valid(versionText);
+    if (!version) {
       return;
     }
 
-    // tslint:disable-next-line: no-console
-    console.error(
-      `npm version ${version} detected. The Angular CLI temporarily requires npm version 6 while upstream issues are addressed.\n\n` +
-      'Please install a compatible version to proceed (`npm install --global npm@6`).\n' +
-      'For additional information and alternative workarounds, please see ' +
-      'https://github.com/angular/angular-cli/issues/19957#issuecomment-775407654',
-    );
-
-    process.exit(3);
+    if (satisfies(version, '>=7 <7.5.6')) {
+      // tslint:disable-next-line: no-console
+      console.warn(
+        `npm version ${version} detected.` +
+          ' When using npm 7 with the Angular CLI, npm version 7.5.6 or higher is recommended.',
+      );
+    }
   } catch {
     // npm is not installed
   }

--- a/tests/legacy-cli/e2e/tests/misc/npm-7.ts
+++ b/tests/legacy-cli/e2e/tests/misc/npm-7.ts
@@ -3,7 +3,7 @@ import { getActivePackageManager } from '../../utils/packages';
 import { ng, npm } from '../../utils/process';
 import { expectToFail } from '../../utils/utils';
 
-const errorText = 'The Angular CLI temporarily requires npm version 6';
+const warningText = 'npm version 7.5.6 or higher is recommended';
 
 export default async function() {
   // Only relevant with npm as a package manager
@@ -18,53 +18,62 @@ export default async function() {
 
   const currentDirectory = process.cwd();
   try {
-    // Install version 7.x
-    await npm('install', '--global', 'npm@7');
+    // Install version >=7.5.6
+    await npm('install', '--global', 'npm@>=7.5.6');
 
-    // Ensure `ng add` exits and shows npm error
-    const { message: stderrAdd } = await expectToFail(() => ng('add'));
-    if (!stderrAdd.includes(errorText)) {
-      throw new Error('ng add expected to show npm version error.');
+    // Ensure `ng update` does not show npm warning
+    const { stderr: stderrUpdate1 } = await ng('update');
+    if (stderrUpdate1.includes(warningText)) {
+      throw new Error('ng update expected to not show npm version warning.');
     }
 
-    // Ensure `ng update` exits and shows npm error
-    const { message: stderrUpdate } = await expectToFail(() => ng('update'));
-    if (!stderrUpdate.includes(errorText)) {
-      throw new Error('ng update expected to show npm version error.');
+    // Install version <7.5.6
+    await npm('install', '--global', 'npm@7.4.0');
+
+    // Ensure `ng add` shows npm warning
+    const { message: stderrAdd } = await expectToFail(() => ng('add'));
+    if (!stderrAdd.includes(warningText)) {
+      throw new Error('ng add expected to show npm version warning.');
+    }
+
+    // Ensure `ng update` shows npm warning
+    const { stderr: stderrUpdate2 } = await ng('update');
+    if (!stderrUpdate2.includes(warningText)) {
+      throw new Error('ng update expected to show npm version warning.');
     }
 
     // Ensure `ng build` executes successfully
     const { stderr: stderrBuild } = await ng('build');
-    if (stderrBuild.includes(errorText)) {
-      throw new Error('ng build expected to not show npm version error.');
+    if (stderrBuild.includes(warningText)) {
+      throw new Error('ng build expected to not show npm version warning.');
     }
 
-    // Ensure `ng new` exits and shows npm error
+    // Ensure `ng new` shows npm warning
     // Must be outside the project for `ng new`
     process.chdir('..');
     const { message: stderrNew } = await expectToFail(() => ng('new'));
-    if (!stderrNew.includes(errorText)) {
-      throw new Error('ng new expected to show npm version error.');
+    if (!stderrNew.includes(warningText)) {
+      throw new Error('ng new expected to show npm version warning.');
     }
 
-    // Ensure `ng new --package-manager=npm` exits and shows npm error
+    // Ensure `ng new --package-manager=npm` shows npm warning
     const { message: stderrNewNpm } = await expectToFail(() => ng('new', '--package-manager=npm'));
-    if (!stderrNewNpm.includes(errorText)) {
-      throw new Error('ng new expected to show npm version error.');
+    if (!stderrNewNpm.includes(warningText)) {
+      throw new Error('ng new expected to show npm version warning.');
     }
 
     // Ensure `ng new --skip-install` executes successfully
     const { stderr: stderrNewSkipInstall } = await ng('new', 'npm-seven-skip', '--skip-install');
-    if (stderrNewSkipInstall.includes(errorText)) {
-      throw new Error('ng new --skip-install expected to not show npm version error.');
+    if (stderrNewSkipInstall.includes(warningText)) {
+      throw new Error('ng new --skip-install expected to not show npm version warning.');
     }
 
     // Ensure `ng new --package-manager=yarn` executes successfully
     // Need an additional npmrc file since yarn does not use the NPM registry environment variable
     await writeFile('.npmrc', 'registry=http://localhost:4873')
     const { stderr: stderrNewYarn } = await ng('new', 'npm-seven-yarn', '--package-manager=yarn');
-    if (stderrNewYarn.includes(errorText)) {
-      throw new Error('ng new --package-manager=yarn expected to not show npm version error.');
+    if (stderrNewYarn.includes(warningText)) {
+      throw new Error('ng new --package-manager=yarn expected to not show npm version warning.');
     }
   } finally {
     // Cleanup extra test projects

